### ISSUE TITLE
codegen doesn't build consistently

### ISSUE THREAD

**Issue by jyjyj** [kobylynskyi/graphql-java-codegen#83]
## Issue Description
I'm using Eclipse and gradle. Whenever I update my graphql schema, I expect to either:

Option 1
1) Refresh Gradle Project
2) Clean project
3) then Refresh project

or

Option 2
1) gradle build

to refresh the generated code (classes and interfaces). My issues are that I haven't been able to find a consistent method of re-generating the Java code after updating the schema. Sometimes I do step (1) and it works and other times it doesn't, same for trying step (2), and at the moment neither of these work. 

## Steps to Reproduce
Can't consistently do it as explained above.

## Expected Result
I'm expecting either step (1) or (2) to correctly re-generate the classes, and hence am flagging it as a bug.

## Actual Result
What I have found as a workaround is to:

1) delete the folder with the generated code
2) run gradle build again
3) refresh gradle project

and the new code comes up. This seems a little strange, and I'm not expecting to need to delete and rebuild every single time I make some changes in my schema.

If I'm wrong and there's actually a proper way to re-generate the Java code consistently, or if delete + rebuild is actually intended, please point me in the right direction! Thanks :)

**Comment by joffrey-bion** [comment]
Thanks for opening this as a separate issue.
As of now, it lacks crucial information and I'm not sure @kobylynskyi or I can be of much help here.

1. Please define "does not work":
   - Is it a Gradle build failure?
   - Is it an IDE error/crash when rebuilding the project from it?
   - Are the expected generated classes actually missing?
   - Are the expected generated classes present but not seen by the IDE?
  These are very different issues, that may or may not be addressed by a fix in this library.

2. Please provide the exact error you get from the build (if it's a Gradle failure), or explain/give a screenshot of your IDE error

3. Please provide an example schema and Gradle build file. Even if we can't reproduce consistently, maybe we can help spotting potential problems.

Without this kind of information, we are absolutely clueless as we don't even know what doesn't work on your machine.

**Comment by jyjyj** [comment]
Hi,

Now that I have separated this from issue #81, I do not see any errors when building gradle, nor does the IDE crash or have any errors. As for whether the classes are missing or present but not seen by the IDE, I have monitored the folder where the code is being generated into, and it does not update when I make a change to the schema.

An example scenario I have (I'm currently still using DataFetcher to build the endpoint): if I have `type Branch { id: String! name: String! }` and change it to `type Branch { branchId: String! name: String! }`, I run gradle refresh and gradle build (both no issues), but when checking the Branch class itself, and the field is still kept as `id` instead of `branchId`.

As I mentioned, I have not been able to consistently get the same errors; will update as and when I do find a related error.

If it may be of use, my build.gradle is as follows:

`plugins {
	id 'org.springframework.boot' version '2.2.2.RELEASE'
	id 'io.spring.dependency-management' version '1.0.8.RELEASE'
	id 'java'
	
	// schema-driven graphql
	id "io.github.kobylynskyi.graphql.codegen" version "1.6.0"
}

/**
 * Generate requests and model from external service
 */
// Automatically generate GraphQL code on project build:
compileJava.dependsOn "graphqlCodegen"
// Add generated sources to your project source sets:
//sourceSets.main.java.srcDir "$buildDir/generated"
sourceSets.main.java.srcDir "$projectDir/src/main/generated"

// no need to define graphql since auto checks resources folder recursively for graphql files
graphqlCodegen {

	// temporary
	graphqlSchemaPaths = [
        "$projectDir/src/main/resources/temp.graphqls".toString()
    ]

    //outputDir = new File("$buildDir/generated")
    outputDir = new File("$projectDir/src/main/generated")
    apiPackageName = "hello.graphql.api"
    modelPackageName = "hello.graphql.model"
    
    // match graphql-datetime implementation
    customTypesMapping = [
    	// doesn't seem to work?
		LocalDateTime: "java.time.LocalDateTime",
		
    	// for extended scalar if using the classic graphql-java
		DateTime: "java.time.OffsetDateTime"
    ]
    generateApis = true
}

group = 'aris'
version = '0.0.1-SNAPSHOT'
sourceCompatibility = '1.8'

repositories {
	mavenCentral()
}

dependencies {
	implementation 'com.graphql-java:graphql-java:11.0' // NEW
    implementation 'com.graphql-java:graphql-java-spring-boot-starter-webmvc:1.0' // NEW
    implementation 'com.google.guava:guava:26.0-jre' // NEW
	implementation 'org.springframework.boot:spring-boot-starter-web'
    testImplementation 'org.springframework.boot:spring-boot-starter-test'
    
    compile 'com.graphql-java-kickstart:graphql-java-tools:6.0.2'
    
    compileOnly "org.projectlombok:lombok:1.18.8"
    annotationProcessor "org.projectlombok:lombok:1.18.8"
	
    // compile all local jar files
    compile fileTree(dir: 'lib', include: ['*.jar'])
    
    // https://github.com/donbeave/graphql-java-datetime
    implementation("com.zhokhov.graphql:graphql-datetime-spring-boot-starter:1.9.0")
    
    // extended scalar if still using the classic graphql-java way
    // https://github.com/graphql-java/graphql-java-extended-scalars
    compile 'com.graphql-java:graphql-java-extended-scalars:1.0'

	// when doing offline? how to map all the compileJava stuff
	//classpath "io.github.kobylynskyi.graphql.codegen:graphql-codegen-gradle-plugin:1.6.0"
}

test {
	useJUnitPlatform()
}
`

Thank you! Will update if I am able to reproduce any errors.

**Comment by joffrey-bion** [comment]
Thanks for the additional info!

If you make changes to the schema, run `gradle build` and then don't see any changes to the files in `src/main/generated`, then it is indeed a bug IMO.

When this happens, do you see in the Gradle log whether the `graphqlCodegen` task is marked `UP-TO-DATE` or `FROM_CACHE`? This could be the reason why the classes are not generated.
That may mean the inputs/outputs of the task are not properly declared, and would require some fix in the plugin.

If you remove `graphqlSchemaPaths` from your task config, it should use all schemas in `src/main/resources` by default. Maybe these task inputs are computed differently.
Do you see the same issue when doing this?

**Comment by jyjyj** [comment]
Hi,

I managed to see the following log with `gradle build --stacktrace`:

`Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':compileJava'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.lambda$executeIfValid$1(ExecuteActionsTaskExecuter.java:205)
        at org.gradle.internal.Try$Failure.ifSuccessfulOrElse(Try.java:263)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:203)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:184)
        at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:114)
        at org.gradle.api.internal.tasks.execution.FinalizePropertiesTaskExecuter.execute(FinalizePropertiesTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.ResolveTaskExecutionModeExecuter.execute(ResolveTaskExecutionModeExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:57)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:56)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
        at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:41)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:372)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:359)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:352)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$BuildOperationAwareExecutionAction.execute(DefaultTaskExecutionGraph.java:338)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.lambda$run$0(DefaultPlanExecutor.java:127)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.execute(DefaultPlanExecutor.java:191)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.executeNextNode(DefaultPlanExecutor.java:182)
        at org.gradle.execution.plan.DefaultPlanExecutor$ExecutorWorker.run(DefaultPlanExecutor.java:124)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
Caused by: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed; see the compiler error output for details.
        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:56)
        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:39)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.delegateAndHandleErrors(NormalizingJavaCompiler.java:100)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:52)
        at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:38)
        at org.gradle.api.internal.tasks.compile.AnnotationProcessorDiscoveringCompiler.execute(AnnotationProcessorDiscoveringCompiler.java:51)
        at org.gradle.api.internal.tasks.compile.AnnotationProcessorDiscoveringCompiler.execute(AnnotationProcessorDiscoveringCompiler.java:37)
        at org.gradle.api.internal.tasks.compile.CleaningJavaCompiler.execute(CleaningJavaCompiler.java:53)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalCompilerFactory.lambda$createRebuildAllCompiler$0(IncrementalCompilerFactory.java:98)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalResultStoringCompiler.execute(IncrementalResultStoringCompiler.java:60)
        at org.gradle.api.internal.tasks.compile.incremental.IncrementalResultStoringCompiler.execute(IncrementalResultStoringCompiler.java:44)
        at org.gradle.api.internal.tasks.compile.CompileJavaBuildOperationReportingCompiler$2.call(CompileJavaBuildOperationReportingCompiler.java:59)
        at org.gradle.api.internal.tasks.compile.CompileJavaBuildOperationReportingCompiler$2.call(CompileJavaBuildOperationReportingCompiler.java:51)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.api.internal.tasks.compile.CompileJavaBuildOperationReportingCompiler.execute(CompileJavaBuildOperationReportingCompiler.java:51)
        at org.gradle.api.tasks.compile.JavaCompile.performCompilation(JavaCompile.java:211)
        at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:182)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
        at org.gradle.api.internal.project.taskfactory.IncrementalInputsTaskAction.doExecute(IncrementalInputsTaskAction.java:32)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:42)
        at org.gradle.api.internal.project.taskfactory.AbstractIncrementalTaskAction.execute(AbstractIncrementalTaskAction.java:25)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:28)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$3.run(ExecuteActionsTaskExecuter.java:568)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:402)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:394)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:92)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.run(DelegatingBuildOperationExecutor.java:31)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:553)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:536)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.access$300(ExecuteActionsTaskExecuter.java:109)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.executeWithPreviousOutputFiles(ExecuteActionsTaskExecuter.java:276)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$TaskExecution.execute(ExecuteActionsTaskExecuter.java:265)
        at org.gradle.internal.execution.steps.ExecuteStep.lambda$execute$0(ExecuteStep.java:32)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:32)
        at org.gradle.internal.execution.steps.ExecuteStep.execute(ExecuteStep.java:26)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:67)
        at org.gradle.internal.execution.steps.CleanupOutputsStep.execute(CleanupOutputsStep.java:36)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:49)
        at org.gradle.internal.execution.steps.ResolveInputChangesStep.execute(ResolveInputChangesStep.java:34)
        at org.gradle.internal.execution.steps.CancelExecutionStep.execute(CancelExecutionStep.java:43)
        at org.gradle.internal.execution.steps.TimeoutStep.executeWithoutTimeout(TimeoutStep.java:73)
        at org.gradle.internal.execution.steps.TimeoutStep.execute(TimeoutStep.java:54)
        at org.gradle.internal.execution.steps.CatchExceptionStep.execute(CatchExceptionStep.java:34)
        at org.gradle.internal.execution.steps.CreateOutputsStep.execute(CreateOutputsStep.java:44)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:54)
        at org.gradle.internal.execution.steps.SnapshotOutputsStep.execute(SnapshotOutputsStep.java:38)
        at org.gradle.internal.execution.steps.BroadcastChangingOutputsStep.execute(BroadcastChangingOutputsStep.java:49)
        at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:44)
        at org.gradle.internal.execution.steps.StoreExecutionStateStep.execute(StoreExecutionStateStep.java:33)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:38)
        at org.gradle.internal.execution.steps.RecordOutputsStep.execute(RecordOutputsStep.java:24)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.executeBecause(SkipUpToDateStep.java:92)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.lambda$execute$0(SkipUpToDateStep.java:85)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:55)
        at org.gradle.internal.execution.steps.SkipUpToDateStep.execute(SkipUpToDateStep.java:39)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:76)
        at org.gradle.internal.execution.steps.ResolveChangesStep.execute(ResolveChangesStep.java:37)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:36)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsFinishedStep.execute(MarkSnapshottingInputsFinishedStep.java:26)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:94)
        at org.gradle.internal.execution.steps.ResolveCachingStateStep.execute(ResolveCachingStateStep.java:49)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:79)
        at org.gradle.internal.execution.steps.CaptureStateBeforeExecutionStep.execute(CaptureStateBeforeExecutionStep.java:53)
        at org.gradle.internal.execution.steps.ValidateStep.execute(ValidateStep.java:74)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.lambda$execute$2(SkipEmptyWorkStep.java:78)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:78)
        at org.gradle.internal.execution.steps.SkipEmptyWorkStep.execute(SkipEmptyWorkStep.java:34)
        at org.gradle.internal.execution.steps.legacy.MarkSnapshottingInputsStartedStep.execute(MarkSnapshottingInputsStartedStep.java:39)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:40)
        at org.gradle.internal.execution.steps.LoadExecutionStateStep.execute(LoadExecutionStateStep.java:28)
        at org.gradle.internal.execution.impl.DefaultWorkExecutor.execute(DefaultWorkExecutor.java:33)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeIfValid(ExecuteActionsTaskExecuter.java:192)
        ... 30 more`

I didn't see a `UP-TO-DATE` or `FROM_CACHE` (might have missed it as I was looking through), but there is some line in the stacktrace that contains the lines `at org.gradle.internal.execution.steps.CacheStep.executeWithoutCache(CacheStep.java:159)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:72)
        at org.gradle.internal.execution.steps.CacheStep.execute(CacheStep.java:43)`, so was wondering if that counts as taking from cache? I have saved my schema whenever I made changes.

I was able, however, to make the gradle build work when running `gradle build --scan` instead. Not sure what the difference is exactly though, since it says `Run with --scan to get full insights.` so I'm not sure why it would be any different from a normal gradle build. This also occurs when I remove `graphqlSchemaPaths` from the build.gradle file. Confusingly, this also isn't consistent, and I have seen cases where `gradle build --scan` doesn't work as well.

Overall:
- [inconsistent] make changes to schema and run `gradle build` => no changes in Java files generated
- run `gradle build --scan` instead, sometimes works (also inconsistent)

Thanks!

**Comment by kobylynskyi** [comment]
> Execution failed for task ':compileJava'
@jyjyj It would be very helpful if you provide the excerpt from your schemas that refers to the non-compiled classes.

**Comment by jyjyj** [comment]
Hi,

Do you mean this?

`D:\graphql-java\src\main\java\hello\main\tasking\TaskingDataFetchers.java:144: error: cannot find symbol
                                output.setTeamId(teamId);
                                      ^
  symbol:   method setTeamId(String)
  location: variable output of type Assignment
D:\graphql-java\src\main\java\hello\main\tasking\TaskingDataFetchers.java:222: error: cannot find symbol
                        assignment.setTeamId((String) obj.get("assigned_to_team_id"));
                                  ^
  symbol:   method setTeamId(String)
  location: variable assignment of type Assignment
Note: D:\graphql-java\src\main\java\hello\main\helper\HttpClientHelper.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors`

I changed a field in my schema in type `Assignment` from `teamId` to `team` to get this error. Once again, only reproduced by running `gradle build --scan`. Thanks!

**Comment by kobylynskyi** [comment]
Ok, so it looks like the issue is that the plugin does not pick up the changes made in your schemas.
I will look into the issue

**Comment by joffrey-bion** [comment]
@kobylynskyi actually, if the schema changed from `teamId` to `team`, it makes perfect sense. It means the class using `setTeamId()` doesn't compile anymore, which is normal given that the newly generated setter is now `setTeam()`.

**Comment by jyjyj** [comment]
@joffrey-bion Yes I understand the need to manually update the implementations. 

What I meant to highlight was that the expected error doesn't come up with `gradle build`, but only comes up with `gradle build --scan` (it doesn't even seem to do so 100% of the time but I have not been able to consistently reproduce). I thought `--scan` was basically the same but showing more insights. Just wondering if this was a bug, thanks.

**Comment by joffrey-bion** [comment]
The `--scan` option only makes gradle publish an in-depth report of what happened during your build (if you say yes when it asks if you accept to upload it).
It should never affect the execution of the build.

My intuition here is that you're trying turning `--scan` on and off, and at the same time you switch from `team` to `teamId` in your schema to simulate changes.
The state with "teamId" matches your code, so the build doesn't fail. The state with "team" doesn't match the code and results in this (expected) build failure. Try switching back and forth your schema without using `--scan` and you should consistently reproduce failure in one state and success in the other.

**Comment by jyjyj** [comment]
@joffrey-bion I have seen the following scenarios so far:

using `gradle build` (no `--scan`)
- change to schema -> generated Java code doesn't update
- Java code update to suit changed schema -> error (since generated code is not updated)

I have checked the generated files each time I rebuild, and the generated Java code is not updated. However, it is updated when I do `gradle build --scan`. As you mention, adding `--scan` should not affect the build, hence I'm finding this weird.

**Comment by joffrey-bion** [comment]
I see, thanks a lot for the updates. This behaviour with `--scan` is really surprising.
Let's focus on the behaviour without the option (generation doesn't occur), at least for now.

I really think there should be something like `UP-TO-DATE` of `FROM_CACHE` or `SKIPPED` next to the generation task.

I looked at the task's code, and it does seem that something is wrong with `graphqlSchemaPaths`. This input is not declared as `@InputFiles` but `@Input`. This means that Gradle will not consider the task inputs to have changed when the *contents* of the schemas change, but only when the *paths* change (which is wrong).

That's what I suspected at the beginning, and we can test whether this is the problem by renaming the file and changing its path in the `graphqlSchemaPaths` config (this makes the path change, so the task input should change in this case).

Now, there is another issue. The way inputs are declared for the `graphqlSchemas` block is also problematic. The `rootDir` is marked as `@InputFile` instead of `@InputDirectory`, so it may cause the same problem (ignoring file contents).
That being said, even the default config suffers from this issue because the fallback to `src/main/resources` is not part of the input of the task per se.

There is definitely some fixes to make in there.

**Comment by kobylynskyi** [comment]
@jyjyj, I was able to reproduce it by changing the schema and executing just `gradle build` (I always did `gradle clean build` before).
The problem was that Gradle marks `graphqlCodegen` task always as **UP-TO-DATE** because it did not correctly process InputFiles..

Until the fix is released, you can do **one of the following**:
* `gradle clean build`
* `gradle build --rerun-tasks`
* Adding the following property to `graphqlCodegen` task:
```groovy
graphqlCodegen {
    ...
    outputs.upToDateWhen { false }
}
```
Thank you for raising this issue!

**Comment by kobylynskyi** [comment]
Te fix will be released in 1.7.0 soon. Stay tuned! :)
